src/dataflow/src/render/join/delta_join.rs

```diff
@@ -227,18 +211,24 @@ where
                                         .expect("Failed to render update stream").0.enter(inner).enter_region(region)
                                 };
 
+                                let initial_closure = join_build_state.extract_closure();
+
                                 // Apply what `closure` we are able to, and record any errors.
-                                let (mut update_stream, errs) = update_stream.flat_map_fallible({
-                                    let mut datums = DatumVec::new();
-                                    let mut row_packer = RowPacker::new();
-                                    move |row| {
-                                        let temp_storage = RowArena::new();
-                                        let mut datums_local = datums.borrow_with(&row);
-                                        // TODO(mcsherry): re-use `row` allocation.
-                                        closure.apply(&mut datums_local, &temp_storage, &mut row_packer).transpose()
-                                    }
-                                });
-                                region_errs.push(errs.map(DataflowError::from));
+                                if !initial_closure.is_identity() {
```

Comment:
this if statement seems like the other functional change (also in the other join implementation as well) - is that correct?

Reply:
Those are the ones I was thinking of yes.